src/composables/auth.ts

```diff
@@ -99,22 +116,32 @@ export const useAuth = createCustomScopedComposable(() => {
     isAuthenticated.value = false;
   }
 
-  async function checkUserAuth() {
+  /**
+   * Open biometric login or password login modal depending on the environment settings.
+   */
+  async function checkUserAuth(): Promise<any> {
     if (!isAuthenticating.value && !isAuthenticated.value && isLoggedIn.value) {
       isAuthenticating.value = true;
       if (IS_MOBILE_APP) {
         if (isBiometricLoginEnabled.value && await checkBiometricLoginAvailability()) {
-          await openModal(MODAL_SECURE_LOGIN);
+          await openModal(MODAL_BIOMETRIC_LOGIN);
         }
       } else if (!encryptionKey.value) {
         await openPasswordLoginModal();
       }
-      // wait before resetting isAuthenticated so that app doesn't register a false app resume event
-      await new Promise((resolve) => setTimeout(resolve, 500));
+
+      // Wait before resetting isAuthenticated so that app doesn't register a false app resume event
+      await sleep(500);
+
       isAuthenticating.value = false;
     }
   }
 
+  async function lockWallet() {
```

Comment:
My attempt for solving this comment: https://github.com/superhero-com/superhero-wallet/pull/3231#discussion_r1764753376